compiler/runtime/OMRRelocationTarget.cpp

```diff
@@ -0,0 +1,58 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+
+#include "env/FrontEnd.hpp"
+#include "control/Options.hpp"
+#include "control/Options_inlines.hpp"
+#include "env/jittypes.h"
+#include "runtime/TRRelocationRecord.hpp"
+#include "runtime/TRRelocationRuntime.hpp"
+#include "runtime/TRRelocationTarget.hpp"
+
+bool OMR::RelocationTarget::isOrderedPairRelocation(TR::RelocationRecord *reloRecord, TR::RelocationTarget *reloTarget)
+   {
+   switch (reloRecord->type(reloTarget))
+      {
+      default:
+         return false;
+      }
+   return false;
+   }
+
+TR::RelocationTarget* 
+OMR::RelocationTarget::self()
+   {
+   return static_cast<TR::RelocationTarget*>(this);
+   }
+
+uint8_t *
+OMR::RelocationTarget::loadAddress(uint8_t *reloLocation)
+   {
+   return  self()->loadPointer(reloLocation);
+   }
+
+void
+OMR::RelocationTarget::storeAddress(uint8_t *address, uint8_t *reloLocation)
```

Comment:
Do we really need both `storeAddress` and `storePointer` ?

Reply:
Thank you for the comment! It is promoted for backwards compatibility, originally, I was trying to bring in as much code that could be reused outside of Java context into OMR as possible. Would you suggest removing the redundant methods?

Reply:
I have removed the abstraction, thank you for suggestion.